src/sentry/incidents/endpoints/serializers/workflow_engine_data_condition.py

```diff
@@ -0,0 +1,109 @@
+from collections import defaultdict
+from typing import DefaultDict
+
+from sentry.api.serializers import Serializer, serialize
+from sentry.incidents.endpoints.serializers.workflow_engine_action import (
+    WorkflowEngineActionSerializer,
+)
+from sentry.incidents.endpoints.utils import translate_threshold
+from sentry.incidents.models.alert_rule import (
+    AlertRuleThresholdType,
+    AlertRuleTrigger,
+    AlertRuleTriggerAction,
+)
+from sentry.workflow_engine.models import (
+    Action,
+    ActionAlertRuleTriggerAction,
+    AlertRuleDetector,
+    DataCondition,
+    DataConditionGroupAction,
+    Detector,
+)
+from sentry.workflow_engine.models.data_condition import Condition
+from sentry.workflow_engine.types import DetectorPriorityLevel
+
+
+class WorkflowEngineDataConditionSerializer(Serializer):
+    def get_attrs(self, item_list, user, **kwargs):
+        data_conditions = {item.id: item for item in item_list}
+        data_condition_groups = [
+            data_condition.condition_group for data_condition in data_conditions.values()
+        ]
+        data_condition_group_actions = DataConditionGroupAction.objects.filter(
+            condition_group_id__in=[dcg.id for dcg in data_condition_groups]
+        )
+        aartas = ActionAlertRuleTriggerAction.objects.filter(
+            action__in=[dcga.action.id for dcga in data_condition_group_actions]
+        )
+        rule_actions = AlertRuleTriggerAction.objects.filter(
+            alert_rule_trigger__in=[aarta.alert_rule_trigger_action.id for aarta in aartas]
+        ).order_by("id")
+
+        actions = Action.objects.filter(
+            id__in=[dcga.action_id for dcga in data_condition_group_actions]
+        ).order_by("id")
+        kwargs = {}  # maybe pass something to help lookup the ActionAlertRuleTriggerAction?
+        serialized_actions = serialize(
+            list(actions), user, WorkflowEngineActionSerializer(), **kwargs
+        )
```

Comment:
Trying to figure out the best way to be able to get the correct `ActionAlertRuleTriggerAction` for the `alertRuleTriggerId` in the `WorkflowEngineActionSerializer`, would it be bad practice to pass the ID in kwargs?